forwardproxy.go

```diff
@@ -99,25 +103,27 @@ func (fp *ForwardProxy) portIsAllowed(port string) bool {
 	return isAllowed
 }
 
-// Copies data r1->w1 and r2->w2, flushes as needed, and returns when both streams are done.
-func dualStream(w1 io.Writer, r1 io.Reader, w2 io.Writer, r2 io.Reader) error {
-	errChan := make(chan error)
+// Copies data targetWriter->clientReader and clientWriter->targetReader, and flushes as needed
+// Returns when both clientWriter-> targetReader stream is done. Caddy should finish writing
+// targetWriter -> clientReader.
+func dualStream(targetWriter io.WriteCloser, clientReader io.ReadCloser,
```

Comment:
`WriteCloser` seems wrong here, since we never call `Close`

Reply:
`ReadCloser`s are also wrong, we don't call close on them either.
dualStream function is no longer asymmetric, and this argument could help to catch errors if arguments are swapped.